src/Tfe.NetClient/Tfe.NetClient.csproj

```diff
@@ -30,5 +30,6 @@
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
     <PackageReference Include="System.Text.Json" Version="4.7.2" />
+    <PackageReference Include="Microsoft.CSharp" Version="4.7.0" />
```

Comment:
Is this reference really needed?

Reply:
I'm afraid so - it's a [bug](https://mikaelkoskinen.net/post/net-core-standard-missing-compiler-required-member) with `netstandard` projects using `dynamic` params - without it the project won't build.

Reply:
As an alternative, I could replace the usage of dynamic with an empty interface, but that's not very good practice.

Reply:
Ok I missed the use of dynamic in some of the Attribute properties. What we should do is create a class with the properties as defined here: https://www.terraform.io/docs/cloud/api/team-access.html#request-body and use it instead of dynamics therefore removing  the need of this package reference

Reply:
So the attributes in the request body are the problem - if `access` is _not_ set to "custom" and any of the other properties exist, terraform responds with a 422 - so I worked around it with `TeamWorkspacePermissions` and `TeamWorkspaceCustomPermissions`. using Automapper or an empty interface were my other thoughts for the parameter, but sprung for `dynamic`, with the `is` check in the constructor trying to maintain some semblance of type safety there. happy to try out another method though, what do you think?

Reply:
I just updated the project to use `System.Text.Json` version 5.0.1 and latest version of C# so now you can try declaring the properties as nullable and use the `JsonIgnore` attribute with the `WhenWritingNull` condition:

``` csharp
        /// <summary>
        /// Access
        /// </summary>
        /// <value></value>
        [JsonPropertyName("access")]
        public string Access { get; set; }

#nullable enable
        /// <summary>
        /// Runs
        /// </summary>
        /// <value></value>
        [JsonPropertyName("runs")]
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string? Runs { get; set; }

        /// <summary>
        /// Variables
        /// </summary>
        /// <value></value>
        [JsonPropertyName("variables")]
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string? Variables { get; set; }

        /// <summary>
        /// StateVersions
        /// </summary>
        /// <value></value>
        [JsonPropertyName("state-versions")]
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string? StateVersions { get; set; }

        /// <summary>
        /// SentinelMocks
        /// </summary>
        /// <value></value>
        [JsonPropertyName("sentinel-mocks")]
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string? SentinelMocks { get; set; }

        /// <summary>
        /// WorkspaceLocking
        /// </summary>
        /// <value></value>
        [JsonPropertyName("workspace-locking")]
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public bool? WorkspaceLocking { get; set; }
#nullable disable
```

If it works the only one attributes class is needed.

Reply:
That worked a treat, thank you for the tip! Hope the update is suitable.

Reply:
LGTM